src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -81,7 +87,7 @@ public function validate($value, Constraint $constraint)
         $host = substr($value, strpos($value, '@') + 1);
 
         // Check for host DNS resource records
```

Comment:
This could be a little more DRY if there were just a single `if (!$this->preventDNSLookups)` conditional wrapping both the MX section and the hostname section.  It would also keep the original `if` statements uncluttered by this new logic.